cli/src/utils/type_example.rs

```diff
@@ -0,0 +1,380 @@
+        };
+
+        let one_example = {
+            let mut values = Vec::with_capacity(self.len as usize);
+            for _ in 0..self.len as usize {
+                values.push(first_value_example.clone());
+            }
+            scale_value::Composite::<()>::Unnamed(values)
+        };
+        Ok(vec![one_example])
+    }
+
+    fn upcast(self_value: Self::Value) -> Value {
+        Value {
+            value: ValueDef::Composite(self_value),
+            context: (),
+        }
+    }
+}
+
+impl TypeExample for TypeDefTuple<PortableForm> {
+    type Value = scale_value::Composite<()>;
+
+    fn type_example(&self, registry: &PortableRegistry) -> color_eyre::Result<Vec<Self::Value>> {
+        // create unnamed fields to use the same logic already used for struct example generation
+        let fields_vector: Vec<Field<PortableForm>> = self
+            .fields
+            .iter()
+            .map(|ty| Field {
+                name: None,
+                ty: *ty,
+                type_name: None,
+                docs: Vec::new(),
+            })
+            .collect();
+        fields_vector.type_example(registry)
+    }
+
+    fn upcast(self_value: Self::Value) -> Value {
+        Value {
+            value: ValueDef::Composite(self_value),
+            context: (),
+        }
+    }
+}
+
+impl TypeExample for Vec<Field<PortableForm>> {
+    type Value = scale_value::Composite<()>;
+
+    fn type_example(&self, registry: &PortableRegistry) -> color_eyre::Result<Vec<Self::Value>> {
+        let all_fields_named = self.iter().all(|f| f.name.is_some());
+        let all_fields_unnamed = self.iter().all(|f| f.name.is_none());
+        // composite apparently has no fields:
+        if all_fields_named && all_fields_unnamed {
+            let one_empty_example = scale_value::Composite::Unnamed(Vec::new());
+            return Ok(vec![one_empty_example]);
+        }
+
+        // composite apparently has mix of named and unnamed fields:
+        if !all_fields_named && !all_fields_unnamed {
+            return Err(eyre!(
+                "combination of named and unnamed fields in compound type"
+            ));
+        }
+
+        // for each field get all the examples the type of that field can offer:
+        let mut field_examples: Vec<(&Field<PortableForm>, Vec<scale_value::Value>)> = Vec::new();
+        for field in self.iter() {
+            let examples = field.ty.id.type_example(registry)?;
+            field_examples.push((field, examples));
+        }
+
+        // Let N be the mininum number of examples any field has.
+        // Return N examples for the Compound type, by choosing the ith example for each of the 0..N examples for that field.
+        let n = field_examples
+            .iter()
+            .map(|(_, examples)| examples.len())
+            .min()
+            .unwrap(); // safe to unwrap, because min could only be None if there are no fields. But in that case we already return an error above.
```

Comment:
in general, we prefer expect over unwrap

```suggestion
            .expect("Iterator is not non-empty checked above; qed")
```